### PR TITLE
remove point2dstamped

### PR DESCRIPTION
**Pull request by ijnek**
Removing Point2DStamped removes the need for sanity checking that frame_id and timestamp matches between the different msgs.

The Point2D will always be in image coordinates, and the timestamp from the PlaneStamped will be used for transforms.

### CODE REVIEW

**Comment by ijnek** [comment]
@Flova when you have the time, would you be able to review this PR? And then I can rebase the other pending ones after. Thanks!

**Comment by Flova** [comment]
See the other PR for discussion regarding this change.

**Comment by Flova** [comment]
Sorry I was on vacation ^^

**Comment by Flova** [comment]
Technically, the PR looks fine, but the redundancy of the time stamp in both the plane and the input points was chosen deliberately.

I get that there are no other frames possible for a given camera info, so the frame_id in the input point is redundant. But it can serve as a sanity check, especially if you have a robot with multiple cameras that the right one is used. Because you need to define the frame at the point where the API is called instead of it being implicitly inferred via the camera info subscription which is published by the camera itself. But I don't have too strong opinions regarding this pattern.

Similarly, but more important, the timestamp might be confusing if it is only set for the plane. I can imagine, that users of the API fail to update the planes header (e.g. they still use the field plane, why should they change something) for every sample and an old time stamp is used. Adding the timestamp of the sample to the input point raises awareness to the fact, that the input is indeed time dependent on the sample of which the point is based on and not the plane which can be constant across long durations. Imo the best API would be one where only the point has the current stamp and the plane only has the plane parameters + a frame where it is attached. But then the PlaneStamped name would be wrong. 

tl;dr: Using only the plane to communicate the timestamp is very confusing for the end user and might lead to timing errors.

**Comment by ijnek** [comment]
I understand your concerns, and I agree with you that associating the timestamp to the plane is confusing.

I'm not a fan of having to send the same information across two different variables to a method, and also don't want to add sanity checks.

As an alternative, what do you think of changing the methods in ipm_library/ipm.py by ditching the Stamped messages:

    def map_point(
            self,
            plane: shape_msgs.msg.Plane,
            point: vision_msgs.msg.Point2D,
            time: Time = builtin_interfaces.msg.Time(),
            plane_frame_id: Optional[str] = None,  # If None, use the camera frame
            output_frame_id: Optional[str] = None  # If None, use the camera frame) -> PointStamped:

    def map_points(
            self,
            plane: shape_msgs.msg.Plane,
            points: np.ndarray,
            stamp: Time = builtin_interfaces.msg.Time(),
            plane_frame_id: Optional[str] = None,  # If None, use the camera frame
            output_frame: Optional[str] = None  # If None, use the camera frame) -> np.ndarray:

**Comment by Flova** [comment]
This sounds good imo! :+1:

**Comment by Flova** [comment]
I will adapt the API accordingly.

**Comment by ijnek** [comment]
@Flova thanks!
